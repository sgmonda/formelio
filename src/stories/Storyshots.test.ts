import initStoryshots from '@storybook/addon-storyshots';

/**
 * This generates and checks jest snapshots based on storybook
 */

initStoryshots();
