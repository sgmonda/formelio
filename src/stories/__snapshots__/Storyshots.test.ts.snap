// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Single Field > Text Empty And Non Required 1`] = `
Array [
  <div
    className="formelio"
  >
    <form
      id="myformid"
    >
      <div
        className="group"
        style={
          Object {
            "marginLeft": "0em",
          }
        }
      >
        <div
          className="fieldWrapper"
          style={
            Object {
              "flexBasis": "100%",
            }
          }
        >
          <div
            className="field"
          >
            <input
              className=""
              id="myformid#myfield"
              name="myfield"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "borderColor": "inherit",
                  "color": "inherit",
                }
              }
              type="text"
              value=""
            />
            <label
              className="isEmpty"
              htmlFor="myformid#myfield"
              style={
                Object {
                  "color": "inherit",
                }
              }
            >
              <span
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                 
                myfield
                 
              </span>
            </label>
          </div>
        </div>
      </div>
    </form>
  </div>,
  <pre>
    {
  "isValid": false
}
  </pre>,
]
`;

exports[`Storyshots Single Field > Text Empty And Non Required With Help 1`] = `
Array [
  <div
    className="formelio"
  >
    <form
      id="myformid"
    >
      <div
        className="group"
        style={
          Object {
            "marginLeft": "0em",
          }
        }
      >
        <div
          className="fieldWrapper"
          style={
            Object {
              "flexBasis": "100%",
            }
          }
        >
          <div
            className="field"
          >
            <input
              className="hasHint"
              id="myformid#myfield"
              name="myfield"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "borderColor": "inherit",
                  "color": "inherit",
                }
              }
              type="text"
              value=""
            />
            <label
              className="isEmpty"
              htmlFor="myformid#myfield"
              style={
                Object {
                  "color": "inherit",
                }
              }
            >
              <span
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                 
                myfield
                 
              </span>
            </label>
            <div
              className="icon"
              style={
                Object {
                  "color": "inherit",
                }
              }
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-question-circle fa-w-16 "
                data-icon="question-circle"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={
                  Object {
                    "color": "inherit",
                    "fontSize": "small",
                    "paddingBottom": "0.1em",
                  }
                }
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>,
  <pre>
    {
  "isValid": false
}
  </pre>,
]
`;

exports[`Storyshots Single Field > Text Empty And Non Required With Icon 1`] = `
Array [
  <div
    className="formelio"
  >
    <form
      id="myformid"
    >
      <div
        className="group"
        style={
          Object {
            "marginLeft": "0em",
          }
        }
      >
        <div
          className="fieldWrapper"
          style={
            Object {
              "flexBasis": "100%",
            }
          }
        >
          <div
            className="field"
          >
            <input
              className=""
              id="myformid#myfield"
              name="myfield"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "borderColor": "inherit",
                  "color": "inherit",
                }
              }
              type="text"
              value=""
            />
            <label
              className="isEmpty"
              htmlFor="myformid#myfield"
              style={
                Object {
                  "color": "inherit",
                }
              }
            >
              <span
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-user fa-w-14 "
                  data-icon="user"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={
                    Object {
                      "color": "inherit",
                      "fontSize": "small",
                      "paddingBottom": "0.1em",
                    }
                  }
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                 
                myfield
                 
              </span>
            </label>
          </div>
        </div>
      </div>
    </form>
  </div>,
  <pre>
    {
  "isValid": false
}
  </pre>,
]
`;

exports[`Storyshots Single Field > Text Empty And Non Required With Icon And Help 1`] = `
Array [
  <div
    className="formelio"
  >
    <form
      id="myformid"
    >
      <div
        className="group"
        style={
          Object {
            "marginLeft": "0em",
          }
        }
      >
        <div
          className="fieldWrapper"
          style={
            Object {
              "flexBasis": "100%",
            }
          }
        >
          <div
            className="field"
          >
            <input
              className="hasHint"
              id="myformid#myfield"
              name="myfield"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "borderColor": "inherit",
                  "color": "inherit",
                }
              }
              type="text"
              value=""
            />
            <label
              className="isEmpty"
              htmlFor="myformid#myfield"
              style={
                Object {
                  "color": "inherit",
                }
              }
            >
              <span
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-user fa-w-14 "
                  data-icon="user"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={
                    Object {
                      "color": "inherit",
                      "fontSize": "small",
                      "paddingBottom": "0.1em",
                    }
                  }
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                 
                myfield
                 
              </span>
            </label>
            <div
              className="icon"
              style={
                Object {
                  "color": "inherit",
                }
              }
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-question-circle fa-w-16 "
                data-icon="question-circle"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={
                  Object {
                    "color": "inherit",
                    "fontSize": "small",
                    "paddingBottom": "0.1em",
                  }
                }
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>,
  <pre>
    {
  "isValid": false
}
  </pre>,
]
`;

exports[`Storyshots Single Field > Text Empty And Required 1`] = `
Array [
  <div
    className="formelio"
  >
    <form
      id="myformid"
    >
      <div
        className="group"
        style={
          Object {
            "marginLeft": "0em",
          }
        }
      >
        <div
          className="fieldWrapper"
          style={
            Object {
              "flexBasis": "100%",
            }
          }
        >
          <div
            className="field"
          >
            <input
              className=""
              id="myformid#myfield"
              name="myfield"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              style={
                Object {
                  "borderColor": "inherit",
                  "color": "inherit",
                }
              }
              type="text"
              value=""
            />
            <label
              className="isEmpty"
              htmlFor="myformid#myfield"
              style={
                Object {
                  "color": "inherit",
                }
              }
            >
              <span
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                 
                myfield
                 
                <span>
                  *
                </span>
              </span>
            </label>
          </div>
        </div>
      </div>
    </form>
  </div>,
  <pre>
    {
  "isValid": false
}
  </pre>,
]
`;

exports[`Storyshots Single Field > Text Empty And Required With Help 1`] = `
Array [
  <div
    className="formelio"
  >
    <form
      id="myformid"
    >
      <div
        className="group"
        style={
          Object {
            "marginLeft": "0em",
          }
        }
      >
        <div
          className="fieldWrapper"
          style={
            Object {
              "flexBasis": "100%",
            }
          }
        >
          <div
            className="field"
          >
            <input
              className="hasHint"
              id="myformid#myfield"
              name="myfield"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              style={
                Object {
                  "borderColor": "inherit",
                  "color": "inherit",
                }
              }
              type="text"
              value=""
            />
            <label
              className="isEmpty"
              htmlFor="myformid#myfield"
              style={
                Object {
                  "color": "inherit",
                }
              }
            >
              <span
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                 
                myfield
                 
                <span>
                  *
                </span>
              </span>
            </label>
            <div
              className="icon"
              style={
                Object {
                  "color": "inherit",
                }
              }
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-question-circle fa-w-16 "
                data-icon="question-circle"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={
                  Object {
                    "color": "inherit",
                    "fontSize": "small",
                    "paddingBottom": "0.1em",
                  }
                }
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>,
  <pre>
    {
  "isValid": false
}
  </pre>,
]
`;

exports[`Storyshots Single Field > Text Non Empty And Non Required 1`] = `
Array [
  <div
    className="formelio"
  >
    <form
      id="myformid"
    >
      <div
        className="group"
        style={
          Object {
            "marginLeft": "0em",
          }
        }
      >
        <div
          className="fieldWrapper"
          style={
            Object {
              "flexBasis": "100%",
            }
          }
        >
          <div
            className="field"
          >
            <input
              className=""
              id="myformid#myfield"
              name="myfield"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "borderColor": "inherit",
                  "color": "inherit",
                }
              }
              type="text"
              value="Hello, world"
            />
            <label
              className=""
              htmlFor="myformid#myfield"
              style={
                Object {
                  "color": "inherit",
                }
              }
            >
              <span
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                 
                myfield
                 
              </span>
            </label>
          </div>
        </div>
      </div>
    </form>
  </div>,
  <pre>
    {
  "isValid": false,
  "value": {
    "myfield": "Hello, world"
  }
}
  </pre>,
]
`;
